pkg/trace/writer/sender.go

```diff
@@ -247,6 +250,10 @@ func (s *sender) sendPayload(p *payload) {
 			return
 		}
 		atomic.AddInt32(&s.attempt, 1)
+
+		if atomic.LoadInt32(&s.attempt) >= maxRetries {
+			log.Warnf("Reached maximum number of retries; error: %s", err.Error())
+		}
```

Comment:
The terminology here is slightly misleading. When we say "maximum retries", it usually means that we give up after having reached that value. Here however it's simply a threshold for a warning. So let's use different wording. Perhaps `retryWarnThreshold` instead of `maxRetries`?

```suggestion
		if atomic.AddInt32(&s.attempt)%retryWarningThreshold == 0 {
			log.Warnf("Retried payload %d times, got error: %s", s.attempt, err.Error())
		}
```

Notice that I've replace `LoadInt32` with `AddInt32`. There is no need to add, and then load, since [AddInt32](https://golang.org/pkg/sync/atomic/#AddInt32) returns the new value. Also, it might be better to log the warning every Nth retry, not just N, so I changed to a module. I'd suggest setting the value to `3` instead of `5`.

Reply:
Yes, the naming fits better. 
The only thing - since `s.attempt` specifies an overall count of failed sends - not per payload - so maybe we should warn that we have a number of different consecutive failed payloads. 

Also, since when a payload goes through, `s.attempt` is swapped with itself divided by two until these values are the same, which might diminish chances of getting those warnings through if we stick to modulo 🤔

Reply:
> The only thing - since s.attempt specifies an overall count of failed sends - not per payload

Ah yes, you're right! Good catch. I missed that. I guess our interest is to log when a specific payload has been retried multiple times.

The question comes - is this not going to be overkill? Should we instead just change the log level from Debug to Warning?

Reply:
The only alternative I see is to add a field to the `payload` type called `retries` and use that.